crates/core/tedge/src/cli/connect/cli.rs

```diff
@@ -41,23 +41,23 @@ impl BuildCommand for TEdgeConnectOpt {
         Ok(match self {
             TEdgeConnectOpt::C8y { is_test_connection } => ConnectCommand {
                 config_location: context.config_location.clone(),
-                config_repository: context.config_repository,
+                config_repository: context.config_repository.skip_environment_variables(),
```

Comment:
Why does the `tedge connect` require write access to the config?

This seems to an issue:
1) I don't see what can be the point here.
2) Using  `tegde connect` in combination with $ENV setting is one of the main point of this PR.

Reply:
`tedge connect` updates tedge.toml with updated MQTT settings. It obviously also updates the mosquitto config with the relevant settings, but making that work in the context of thin-edge running in a container requires some sort of persistence for `/etc/tedge` anyway, so I'm not entirely sure what we gain by avoiding using `/etc/tedge/tedge.toml` in that case? @reubenmiller can you provide some insight as to how you're expecting this feature to be used?

Reply:
I'm wondering whether instead of opt-out of environment variables entirely when storing values, we should replace the `store` method with an `update_toml` method that takes an argument `impl FnOnce(Config) -> Config`. The repository calls that with only the TOML config, and saves whatever is returned back to the TOML file. This allows us to better scope the updates to the TOML config in cases like this where we may want to refer to environment variables but also update the persistent configuration.

Reply:
> tedge connect updates tedge.toml with updated MQTT settings.

This comes as a surprise to me. I really don't understand what's the point.

Reply:
> `tedge connect` updates tedge.toml with updated MQTT settings. It obviously also updates the mosquitto config with the relevant settings, but making that work in the context of thin-edge running in a container requires some sort of persistence for `/etc/tedge` anyway, so I'm not entirely sure what we gain by avoiding using `/etc/tedge/tedge.toml` in that case? @reubenmiller can you provide some insight as to how you're expecting this feature to be used?

Idea is to create multiple containers where each container runs one component of thin-edge, e.g. one container for tedge-agent, one for tedge-mapper-c8y, one for mosquitto. Each container will be configured to use an external mqtt broker by setting the mqtt client settings to point to the mosquitto broker (via environment variables). The actual values inside the tedge.toml don't need to store the values set in the environment variables.

The `tedge connect` command will only be used to setup the mosquitto broker bridge settings (or in some cases it will not be used at all, as technically the user can create the bridge settings themselves).